remoulade/state/backends/redis.py

```diff
@@ -0,0 +1,33 @@
+import redis
+
+from ..backend import StateBackend
+
+
+class RedisBackend(StateBackend):
+    def __init__(
+        self,
+        *,
+        namespace="remoulade-state",
+        encoder=None,
+        client=None,
+        url=None,
+        **parameters
+    ):
+        super().__init__(namespace=namespace, encoder=encoder)
+        if url:
+            parameters["connection_pool"] = redis.ConnectionPool.from_url(url)
+        self.client = client or redis.Redis(**parameters)
+
+    def get_state(self, message_id):
```

Comment:
I've not been very clear I think. `get_state` should return a state object according to the superclass.  And set state should get a state object. So you will need to call `from_dict` and `to_dict` inside these functions.

Do you about mypy, it's type checking for python: it would have caught this. Maybe we can add a mypy step in the CI in a later PR. Can you create an issue about this ?

https://mypy.readthedocs.io/en/stable/

Reply:
really cool, thanks